### ISSUE TITLE
Missing cover image when converting into OGG

### ISSUE THREAD

**Issue by xtemp09** [lordmulder/LameXP#23]
I converted a FLAC into OGG and found that cover art is missing.

**Comment by lordmulder** [comment]
The Ogg/Vorbis Encoder (`oggenc2.exe`) does **not** offer a way to pass an artwork image &ndash; that's why.

Maybe artwork could be added, in a separate step, via `vorbiscomment.exe`, but I'll have to see about that...

**Comment by xtemp09** [comment]
Opusenc2 doesn't support art embedding; output APE files lack cover image too. It will require third-party tool(s) to add that feature. Looking through temporary directory, I found that LameXP uses lxp_tag for mp3 metadata.

Might I point at a multifunctional library, like TagLib (http://taglib.github.io), that can kill several birds with one stone?

**Comment by lordmulder** [comment]
Nope, 'tag.exe' is used only to embed is only used to emebed ApeTags 
into Monkey's Audio files. LAME supports adding meta tags to the MP3 it 
creates files just fine.

Since OggEnc2 doesn't support cover artwork, we could use VorbisComment 
to add cover artwork to Ogg/Vorbis files. But VorbisComment doesn't 
provide a way to easily add cover artwork either. It just takes the 
"raw" data. For cover artwork, we would have to put together a 
METADATA_BLOCK_PICTURE structure manually. I will have to look into that.


On 23.08.2015 22:15, xtemp09 wrote:
>
> Opusenc2 doesn't support art embedding; output APE files lack cover 
> image too. It will require third-party tool(s) to add that feature. 
> Looking through temporary directory, I found that LameXP uses lxp_tag 
> for mp3 metadata.
>
> Might I point at a multifunctional library, like TagLib 
> (http://taglib.github.io), that can kill several birds with one stone?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/lordmulder/LameXP/issues/23#issuecomment-133918138>.
>

-- 
https://www.youtube.com/v/VF9UMona74w?autoplay=1
